spark/v3.3/spark-extensions/src/main/antlr/org.apache.spark.sql.catalyst.parser.extensions/IcebergSqlExtensions.g4

```diff
@@ -73,6 +73,13 @@ statement
     | ALTER TABLE multipartIdentifier WRITE writeSpec                                       #setWriteDistributionAndOrdering
     | ALTER TABLE multipartIdentifier SET IDENTIFIER_KW FIELDS fieldList                    #setIdentifierFields
     | ALTER TABLE multipartIdentifier DROP IDENTIFIER_KW FIELDS fieldList                   #dropIdentifierFields
+    | ALTER TABLE multipartIdentifier CREATE BRANCH identifier (AS OF VERSION snapshotId)?  (snapshotRetentionClause)? (RETAIN snapshotRefRetain snapshotRefRetainTimeUnit)?   #createBranch
```

Comment:
feels more natural for the `RETAIN` clause to be before the `snapshotRetentionClause`? Because clauses like:

```
CREATE BRANCH branch AS OF VERSION 12345 
WITH SNAPSHOT RETENTION 2 SNAPSHOTS 2 DAYS
RETAIN 2 DAYS
```

vs 

```
CREATE BRANCH branch AS OF VERSION 12345 
RETAIN 2 DAYS
WITH SNAPSHOT RETENTION 2 SNAPSHOTS 2 DAYS
```

Making `RETAIN` after `WITH SNAPSHOT RETENTION` makes me feel a bit lost about what that clause is doing. By moving it closer to `CREATE BRANCH`, it is clearer that `RETAIN` is describing expiration of the branch.

What do you think?

Reply:
Agreed, after re-reading the syntax I felt this way too. `RETAIN` should be close to the actual branch clause so that association of what the clauses is responsible for (the retention of the reference itself) is there.

Reply:
Sounds reasonable to me.